### PR TITLE
sort_stack in javascript

### PR DESCRIPTION
**Pull request by tbhaxor**
**Fixes issue:** None

**Changes:** Added sort stack in javascript

### CODE REVIEW

File: code/data_structures/src/stack/sort_stack/sort_stack.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+var Stack = function (initStack = [])
```
Comment (arnavb) [inline code/data_structures/src/stack/sort_stack/sort_stack.js]: let let let. In fact, use `es6` [classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes).

File: code/data_structures/src/stack/sort_stack/sort_stack.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{
+    this.stack = initStack;
+    this.top = initStack.length;
+    
+    this.push = function (value) {
```
Comment (arnavb) [inline code/data_structures/src/stack/sort_stack/sort_stack.js]: This stuff becomes a lot easier with classes.

File: code/data_structures/src/stack/sort_stack/sort_stack.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        return this.top;
+    }
+
+    this.peek = function() {
+        return this.stack[this.top - 1];
+    }
+
+    this.pop = function () { 
+        if (this.top === 0) {
+            return "Stack is Empty";
```
Comment (arnavb) [inline code/data_structures/src/stack/sort_stack/sort_stack.js]: Don't mix and match return types (unless undefined). Either use an empty `return;` or throw an exception.

File: code/data_structures/src/stack/sort_stack/sort_stack.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        }
+        this.top--;
```
Comment (arnavb) [inline code/data_structures/src/stack/sort_stack/sort_stack.js]: Preincrement.

Comment (tbhaxor) [reply code/data_structures/src/stack/sort_stack/sort_stack.js]: with due respect, sir its _post decrement_

**Comment by tbhaxor** [comment]
@arnavb  i m new to JavaScript.
I dnt know about what you are talking about ?

should I drop this PR because i can't code what you want or you ll accept it ?

**Comment by arnavb** [comment]
@tbhaxor I'm also new to JS! 😆 But I know of a good feature and you can read and use them from [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes). It is very easy to understand and converting the code should be easy. A class is basically a "blueprint" for objects that are instantiated from it.

**Comment by tbhaxor** [comment]
ohk sir, i am lil busy as in India March is the end of Indian years :D

File: code/data_structures/src/stack/sort_stack/sort_stack.js (hunk 4 above)
Comment (arnavb) [reply code/data_structures/src/stack/sort_stack/sort_stack.js]: @tbhaxor I am asking you to change your code to pre-increment. I know what post-increment is...

Comment (tbhaxor) [reply code/data_structures/src/stack/sort_stack/sort_stack.js]: ohk sir, i apologize

**Comment by danos7777** [comment]
I'm sorry I didn't know how to do that and I really like to learn more about it and I have no clue how to do it. I'm just trying to figure out how you change codes and make my phone unhackable

**Pull request closed without merging**